server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -305,6 +307,16 @@ public Set<Reference> droppedColumns() {
         return droppedColumns;
     }
 
+    /**
+     * Get a TranslogIndexer based on this table
+     */
+    public TranslogIndexer getTranslogIndexer() {
+        if (this.translogIndexer == null) {
+            this.translogIndexer = new TranslogIndexer(this);
+        }
+        return this.translogIndexer;
+    }
+
```

Comment:
Is this to piggy back onto the cache? It bothers me a bit to turn the `DocTableInfo` from a mostly data-class to something with more capabilities, but can't put my finger on it and also don't have a better idea. So I guess this is fine, just wanted to mention it in case you had concerns too.

Reply:
Yes, this was the best way I could figure out to ensure that the translog was always in-sync in terms of mappings with the schema.